src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -213,7 +213,14 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 				if (injection.filesToCopy) {
 					for (const f of injection.filesToCopy) {
 						await fs.mkdir(path.dirname(f.dest), { recursive: true });
-						await fs.copyFile(f.source, f.dest);
+						try {
+							await fs.copyFile(f.source, f.dest);
+						} catch {
+							// Swallow error, this should only happen when multiple users are on the same
+							// machine. Since the shell integration scripts rarely change, plus the other user
+							// should be using the same version of the server in this case, assume the script is
```

Comment:
I don't think you can assume that two users are on the same client version of vscode, especially around the time of a release. And if the name is as generic as `/tmp/vscode-zsh/.zshrc` couldn't you have an issue with a single user upgrading too? And insiders vs stable?

Reply:
@roblourens I think you're right about multiple versions of the server running at once, though maybe we should notify if we don't when a newer version is installed on the same machine? I didn't want to get into the business of leaving a bunch of per version files around. The impact of not writing this file is quite small as the script doesn't change often and even if it did, it likely wouldn't cause problems and they're easily resolved (update to latest).

We should include the product name in the folder, this was an oversight.

Reply:
> they're easily resolved (update to latest).

When you update, how does it end up actually using a newer version of this file? Seems like the oldest will stick around forever?

Reply:
@roblourens oh, I thought this was related to a lock on the file but it's because a user cannot modify another user's file? I'll reopen the issue then

Reply:
That's my assumption - I haven't tried to break it